src/client/camera.cpp

```diff
@@ -316,16 +316,29 @@ void Camera::update(LocalPlayer* player, f32 frametime, f32 tool_reload_ratio)
 	// Get player position
 	// Smooth the movement when walking up stairs
 	v3f old_player_position = m_playernode->getPosition();
-	v3f player_position = player->getPosition();
+	v3f player_position = player->getEyePosition();;
+	v3f parent_rotation(0, 0, 0);
 
 	f32 yaw = player->getYaw();
 	f32 pitch = player->getPitch();
 
 	// This is worse than `LocalPlayer::getPosition()` but
 	// mods expect the player head to be at the parent's position
 	// plus eye height.
-	if (player->getParent())
-		player_position = player->getParent()->getPosition();
+	if (player->getParent()) {
+		GenericCAO *parent = dynamic_cast<GenericCAO *>(player->getParent());
+		int parent_id;
+		v3f attach_pos;
+		v3f attach_rot;
+		std::string bone;
+		bool force_visible;
+		GenericCAO *player_cao = player->getCAO();
+		player_cao->getAttachment(&parent_id, &bone, &attach_pos, &attach_rot, &force_visible);
+		parent_rotation = parent->getRotation() - attach_rot;
```

Comment:
I don't think this is correct in general. You want to ~~undo~~ apply `attach_rot` here, but you can't do that by subtracting Euler angles. You'd have to use quaternions: First convert to quaternions, then invert the quaternion for the attachment rotation (different conventions for rotation orientation?), then multiply that with the quaternion for the parent rotation, then convert back to Euler angles (and be careful about radians vs degrees).

Reply:
I tried to do this:

```
     player_cao->getAttachment(&parent_id, &bone, &attach_pos, &attach_rot, &force_visible);
     core::quaternion quat_rot(parent->getRotation() * core::DEGTORAD);
     quat_rot = quat_rot * core::quaternion(-attach_rot * core::DEGTORAD);
     parent_rotation = parent->getRotation() - attach_rot; // not used here
     quat_rot.toEuler(parent_rotation);
     parent_rotation = parent_rotation * core::RADTODEG;
     attach_pos += player->getEyeOffset();
     parent->getPosRotMatrix().rotateVect(attach_pos);
     player_position = parent->getPosition() + attach_pos;
```

And it stops the camera from following the player's eyes. It follows only parent node rotation and attach_rot makes no difference. I tried also `quat_rot = core::quaternion(-attach_rot * core::DEGTORAD) * quat_rot;`, but without visible effect.

I have debugged it with vectors like:

attach_rot = {X = -5.26318979, Y = 0, Z = 93.7392807}
parent_rotation = {X = 355.990173, Y = 0, Z = 0.400718689} -> after sub with attach_rot

and for this, quaternion gets the result {X = -9.65404415, Y = -5.24941778, Z = 0.41818282} in DEG

Based on https://stackoverflow.com/questions/56246893/how-do-i-rotate-a-quaternion-with-2nd-quaternion-on-its-local-or-world-axes-with

Another try:

```
     player_cao->getAttachment(&parent_id, &bone, &attach_pos, &attach_rot, &force_visible);
     parent_rotation = parent->getRotation() - attach_rot;
     core::quaternion quat_rot(parent_rotation * core::DEGTORAD);
     quat_rot.toEuler(parent_rotation);
     parent_rotation = parent_rotation * core::RADTODEG;
     attach_pos += player->getEyeOffset();
     parent->getPosRotMatrix().rotateVect(attach_pos);
```

Both with/without:
```
if(parent_rotation.X < 0)
    parent_rotation.X += 360;
...
```

And some results as the previous try. Camera positions follow only parent rotation, and `attach_rot` makes no visible effect.

Based on https://irrlicht.sourceforge.io/forum/viewtopic.php?p=285337

Reply:
Finnaly this looks to works better:

```
     core::quaternion quat_rot(parent->getRotation() * core::DEGTORAD);
     quat_rot = quat_rot * core::quaternion(-attach_rot * core::DEGTORAD);
     parent_rotation = parent->getRotation() - attach_rot; // not used here
     quat_rot.toEuler(parent_rotation);
     parent_rotation = parent_rotation * core::RADTODEG;
     quat_rot.normalize();
     v3f eye_offset = player->getEyeOffset();
     quat_rot.getMatrix().inverseRotateVect(eye_offset);
     attach_pos += eye_offset;
     //parent->getPosRotMatrix().rotateVect(attach_pos);
     
     player_position = parent->getPosition() + attach_pos;

```

But camera rotation stops following parent and attachment rotation.

Reply:
Please, check last commit.
This solutions looks to be working well.

i still have to update pich and yawto be relative too and similar. But that code will be delivered from from camera orientation code, I expect.